apiserver/common/controllerconfig.go

```diff
@@ -76,15 +77,44 @@ func (s *controllerStateShim) ControllerInfo(modelUUID string) (addrs []string,
 		return StateControllerInfo(s.State)
 	}
 
-	// Now check any external controllers.
 	ec := state.NewExternalControllers(s.State)
 	info, err := ec.ControllerForModel(modelUUID)
+
+	// the model may have been migrated, if that is the case we try to get the information.
+	// If that is true we try to save the migration controller as an external controller.
+	if errors.IsNotFound(err) {
```

Comment:
This doesn't look like it will work.

External controllers are stored by models consuming remote offers so that once they add the relation they can notify that model of consumption and watch the remote offer for changes.

If you are already trying to locate an _external controller_ for a model, it is unlikely the model was previously _local to this controller_.

Reply:
I am not following 100%. 
Please correct me if my thoughts are wrong here:
- after the model migration this code cannot find the `modelUUID` in the current controller collection anymore  -> returns error.notFound
- Thats why I look into the `migrations` collection to search for the model to see whether it has been migrated.
- If that is the case save this controller into the `externalcontroller` collection

At least thats how I understood it and coded it into the corresponding unit test.

Reply:
Aha. 

I needed to expand above this in the Github UI. I didn't see that looking for external controllers was already a fall-back for checking the local model.

We still have the problem of locating a migrated model _from_ an external controller, but we will have discussions about this next week.